Source/core/FileObserver.h

```diff
@@ -245,42 +249,173 @@ class FileSystemMonitor : public Core::IResource {
+        Context() = delete;
+        Context(const Context&) = delete;
+        Context& operator= (const Context&) = delete;
+
+        Context(const string& pathName)
+            : _adminLock()
+            , _filename(pathName)
+            , _file(INVALID_HANDLE_VALUE)
+            , _clients() {
+            _file = CreateFile(pathName.c_str(),
+                FILE_LIST_DIRECTORY,
+                FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE,
+                NULL,
+                OPEN_EXISTING,
+                FILE_FLAG_BACKUP_SEMANTICS | FILE_FLAG_OVERLAPPED,
+                NULL);
+
+            if (_file != INVALID_HANDLE_VALUE) {        
+                _overlapped.hEvent = CreateEvent(NULL, FALSE, 0, NULL);
+
+                if (ReadDirectoryChangesW(
+                    _file, _buffer, sizeof(_buffer), FALSE,
+                    FILE_NOTIFY_CHANGE_FILE_NAME |
+                    FILE_NOTIFY_CHANGE_DIR_NAME,
+                    NULL, &_overlapped, NULL) == FALSE) {
+                    TRACE_L1("Could not start observing: %s", _filename.c_str());
+                }
+            }
         }
-        ~Worker() override {
+        ~Context() {
+            if (_file != INVALID_HANDLE_VALUE) {
+                ::CloseHandle(_file);
+                if (_overlapped.hEvent != INVALID_HANDLE_VALUE) {
+                    ::CloseHandle(_overlapped.hEvent);
+                }
+            }
         }
 
     public:
-        HANDLE Event() {
-            return(_syncEvent);
+        void Register(ICallback* client) {
+            ASSERT(std::find(_clients.begin(), _clients.end(), client) == _clients.end());
+
+            _clients.push_back(client);
+        }
+        void Unregister(ICallback* client) {
+            ClientList::iterator index (std::find(_clients.begin(), _clients.end(), client));
+
+            ASSERT (index != _clients.end());
+
+            if (index != _clients.end()) {
+                _clients.erase(index);
+            }
         }
-        uint32_t Worker() override {
-        
-            return (Core::infinite);
+        HANDLE Handle() {
+            return (_overlapped.hEvent);
         }
+        bool IsEmpty() const {
+            return (_clients.empty());
+        }
+        void Notify() {
+            DWORD bytes_transferred;
+            if (GetOverlappedResult(_file, &_overlapped, &bytes_transferred, FALSE) != FALSE) {
+                if (ReadDirectoryChangesW(
+                    _file, _buffer, sizeof(_buffer), FALSE,
+                    FILE_NOTIFY_CHANGE_FILE_NAME |
+                    FILE_NOTIFY_CHANGE_DIR_NAME,
+                    NULL, &_overlapped, NULL) == FALSE) {
+
+                    TRACE_L1("Could not start observing: %s", _filename.c_str());
+                }
+
+                for (auto& client : _clients) {
+                    client->Updated();
+                }
+            }
+        }
+
+        /*           
```

Comment:
remove commented out code or do you still need it?

Reply:
Don't know if we still need it, found it with an example on internet. Wanted to store it for reference later on. Can also link the example from which it was taken :-)